draft-ietf-ppm-dap.md

```diff
@@ -2226,14 +2210,48 @@ structure includes the following:
   reports included in the batch, such that the interval's start and duration are
   both multiples of the task's `time_precision` parameter. Note that in the case
   of a time-interval query (see {{batch-mode}}), this interval can be smaller
-  than the one in the corresponding `CollectionReq.query`.
+  than the one in the corresponding `CollectionJobReq.query`.
 
 * `leader_encrypted_agg_share`: The Leader's aggregate share, encrypted to the
   Collector.
 
 * `helper_encrypted_agg_share`: The Helper's aggregate share, encrypted to the
   Collector.
 
+If the Leader finds the `CollectionJobReq` to be valid, it immediately responds
+with HTTP status 201 Created with a body containing a `CollectionJobResp`. The
+`status` field is set to `processing`. The Leader SHOULD include a Retry-After
+header field to suggest a polling interval to the Collector.
+
+After receiving the response to its `CollectionJobReq`, the Collector
+periodically makes HTTP GET requests to the same collection job URI to check on
+the status of the collect job and eventually obtain the result. The Leader
+responds to GET requests with HTTP status 200 and the `status` field reflecting
+the current state of the job. When the collection job is `processing`, the
+response SHOULD include a Retry-After header field to suggest a polling interval
+to the Collector.
+
+The Leader then begins working with the Helper to aggregate the reports
+satisfying the query (or continues this process, depending on the VDAF) as
+described in {{aggregate-flow}}.
+
+The Leader first checks whether it can construct a batch for the
+collection job by applying the requirements in {{batch-validation}}. If so, then
+the Leader obtains the Helper's aggregate share following the aggregate-share
+request flow described in {{collect-aggregate}}. If not, it either aborts the
+collection job or tries again later, depending on which requirement in
+{{batch-validation}} was not met. If the Leader has a pending aggregation job
+that overlaps with the batch and aggregation parameter for the collection job,
+the Leader MUST first complete the aggregation job before proceeding and
+requesting an aggregate share from the Helper. This avoids a race condition
+between aggregation and collection jobs that can yield trivial batch mismatch
+errors.
+
+Once both aggregate shares are successfully obtained, the Leader responds to
+subsequent HTTP GET requests with the `status` field set to `finished` and the
+`Collection` field populated accordingly. The Collector stops polling once
```

Comment:
Replace `XXX` with the relevant section (the Leader has the Helper's encrypted agg share and produces its own encrypted agg share)
```suggestion
`Collection` field populated as constructed as specified in {{XXX}}. The Collector stops polling once
```

Reply:
Sorry, I'm not seeing a section that best fits here. I suppose I can link the reader to the paragraph `The `Collection` structure includes the following` but that's midway through the section. (Maybe this section needs to be broken up a bit?)

Reply:
You're right. Maybe "with the `Collection` field populated with the encrypted aggregate shares".

Reply:
How does the latest look? I broke up the `tls-presentation` definition into two pieces, so the narrative flows a bit better generally. (i.e. "here's what it looks like when processing, now here's what it looks like when finished")

Reply:
It's better if the definition of `Collection` comes before `CollectionJobResp`, since the latter depends on the former.

Reply:
> You're right. Maybe "with the Collection field populated with the encrypted aggregate shares".

I went with this.